packages/template-retail-react-app/app/components/store-locator-modal/store-locator-content.jsx

```diff
@@ -5,21 +5,133 @@
  */
 
-import React from 'react'
+import React, {useState, useContext} from 'react'
 import {useIntl} from 'react-intl'
-import PropTypes from 'prop-types'
+
+// Components
 import {
     Heading,
     Accordion,
     AccordionItem,
-    Box
+    Box,
+    Button
 } from '@salesforce/retail-react-app/app/components/shared/ui'
 import StoresList from '@salesforce/retail-react-app/app/components/store-locator-modal/stores-list'
 import StoreLocatorInput from '@salesforce/retail-react-app/app/components/store-locator-modal/store-locator-input'
-import {STORE_LOCATOR_DISTANCE} from '@salesforce/retail-react-app/app/constants'
 
-const StoreLocatorContent = ({form, submitForm, storesInfo, searchStoresParams}) => {
+// Others
+import {
+    SUPPORTED_STORE_LOCATOR_COUNTRIES,
+    DEFAULT_STORE_LOCATOR_COUNTRY,
+    STORE_LOCATOR_DISTANCE,
+    STORE_LOCATOR_NUM_STORES_PER_LOAD,
+    STORE_LOCATOR_DISTANCE_UNIT
+} from '@salesforce/retail-react-app/app/constants'
+const NUM_STORES_PER_REQUEST_API_MAX = 200
+
+// Hooks
+import {useSearchStores} from '@salesforce/commerce-sdk-react'
+import {useForm} from 'react-hook-form'
+
+import {StoreLocatorContext} from '@salesforce/retail-react-app/app/components/store-locator-modal/index'
+
+const StoreLocatorContent = () => {
+    const {
+        searchStoresParams,
+        setSearchStoresParams,
+        userHasSetManualGeolocation,
+        setUserHasSetManualGeolocation
+    } = useContext(StoreLocatorContext)
     const intl = useIntl()
+    const form = useForm({
+        mode: 'onChange',
+        reValidateMode: 'onChange',
+        defaultValues: {
+            countryCode: userHasSetManualGeolocation ? searchStoresParams.countryCode : '',
+            postalCode: userHasSetManualGeolocation ? searchStoresParams.postalCode : ''
+        }
+    })
+
+    const [numStoresToShow, setNumStoresToShow] = useState(searchStoresParams.limit)
+
+    const {data: searchStoresData, isLoading} = useSearchStores(
+        {
+            parameters: {
+                countryCode: searchStoresParams.latitude
+                    ? undefined
+                    : searchStoresParams.countryCode,
+                postalCode: searchStoresParams.latitude ? undefined : searchStoresParams.postalCode,
+                latitude: searchStoresParams.countryCode ? undefined : searchStoresParams.latitude,
+                longitude: searchStoresParams.countryCode
+                    ? undefined
+                    : searchStoresParams.longitude,
+                locale: intl.locale,
+                maxDistance: STORE_LOCATOR_DISTANCE,
+                limit: NUM_STORES_PER_REQUEST_API_MAX,
+                distanceUnit: STORE_LOCATOR_DISTANCE_UNIT
+            }
+        },
+        {keepPreviousData: true}
+    )
+
+    const storesInfo = isLoading
+        ? undefined
+        : searchStoresData?.data
+        ? searchStoresData?.data.slice(0, numStoresToShow)
+        : []
```

Comment:
nit, chaining ternary operators are really hard to read, would you consider this?
```suggestion
    const storesInfo = isLoading
        ? undefined
        : (searchStoresData?.data.slice(0, numStoresToShow) || [])
```